### PR TITLE
Java 17 & Cleaned up build pipeline

### PR DESCRIPTION
**Pull request by fussel178**
### Summary 

This PR introduces Java 17 support for Telestion and cleans up the configuration for the build pipeline.

### Details 

Due to many deprecations in the Gradle API and the somewhat spooky build configuration of Telestion Core I took a deep-dive into Gradle and multi-project configuration with shared build configurations.
I tried to be as convention-complete to the Gradle docs as possible. Maybe I missed some parts, so be sure to check out the changes I made and mark parts that are missing or incomplete.
To the new configuration approach:

#### Plugin-based configuration

The Gradle docs say that `allprojects` and `subprojects` closures no longer recommended because of the increased configuration time and the weird configuration logic. So I introduced two Gradle plugins that abstract the build logic for a Telestion Core Java Project (plugin named `telestion-java`) and a Telestion Core Java Library Project (plugin named `telestion-java-library`) and an extension to configure these plugins. (An extension is like a data storage that is defined in the project and passed into the plugin during configuration)
These plugins are defined in the new `buildSrc` folder. This folder has a somewhat special meaning in Gradle terms as it's designed to hold plugins for the project. The plugin definitions inside this folder look like the Gradle DSL for the project configurations and that is the trick of this special folder.
These plugins are then applied to every subproject in Telestion Core (e.g. `telestion-api`, `telestion-services`, ...) and thus reduce the duplicate configuration and maintenance overhead.

#### The way to Kotlin

During the creation of this tasks I had a real headache with the dynamic/weak typing of Groovy and decided to switch to Kotlin as configuration language. (same reason as above: better maintenance :stuck_out_tongue_winking_eye:)

#### Dependency catalog

Gradle provides a relatively new feature to merge the definition of same dependencies in different project into one declaration spot. This feature is called dependency catalog and the new catalog for Telestion Core resides in `gradle/libs.versions.toml`.
Simply said, you give a dependency an alias and use this alias throughout the project. So you have only one dependency with the same version installed.
With a view to our new dependency updater [renovate](https://docs.renovatebot.com/) this is useful because the updater supports this kind of dependency definition.

#### Regressions

I removed the `allJavadoc` task that combined the generated Javadoc from all project into one big Javadoc webpage because we currently don't use any of the generated Javadoc and link instead to the Javadoc reader <https://javadoc.io/>. If we decide to include this task again, we need to take a look into the Gradle API because some features in Groovy are not compatible with Java or Kotlin.

### Additional information 

Short and spicy changelog:

- update to Gradle v7.6
- declare all dependencies in `gradle/libs.versions.toml`
- add `buildSrc`
- add `telestion-java` plugin to configure java-based projects
- add `telestion-java-library` plugin to configure java-library projects
- add extension to configure the pretty name and description of the project
- move the entire build, test and deploy pipeline into these plugins
- move all subprojects out of the `:module` subproject
- rewrite all project configurations to use the new plugins
- switch to Kotlin as configuration language
- removed the `alljavadoc` task

### CLA

- [x] I have signed the individual contributor's license agreement **and sent** it to the board of the WüSpace e. V. organization.

### Note

As always, if no one complains I see the PR implicitly accepted after 2 weeks. :wink:

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+## Note:
+##
+## Add dependencies that you want to use in this definition.
+## You can access these after a gradle sync via:
+##
+## dependencies {
+##     implementation(libs.vertx.web)
+## }
```
Comment (pklaschka) [inline gradle/libs.versions.toml:8]: I believe you, but am curious how this works. Given that it's notated as `vertx-web` below (and the package name also is `vertx-web`), how does it infer that from the `.` in `vertx.web`?

File: telestion-api/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// some meta information (important for packaging metadata)
+telestion {
+    prettyName.set("Telestion API")
+    description.set("The essential components for a Telestion Project Application")
+}
+
+tasks.withType<Test>().configureEach {
+    systemProperty("junit.jupiter.displayname.generator.default", "de.wuespace.telestion.api.CamelCase")
+}
+
```
Comment (pklaschka) [inline telestion-api/build.gradle.kts:14]: What does this do?

**Review by pklaschka — APPROVED** [review]
LGTM! Thanks for your contribution! :tada:

File: gradle/libs.versions.toml (hunk 1 above)
Comment (fussel178) [reply gradle/libs.versions.toml:8]: Gradle automatically converts the separators to the `.` separator used in this example.

> When declaring aliases, it’s worth noting that any of the -, _ and . characters can be used as separators, but the generated catalog will have all normalized to .: for example foo-bar as an alias is converted to foo.bar automatically.
> https://docs.gradle.org/current/userguide/platforms.html

File: telestion-api/build.gradle.kts (hunk 2 above)
Comment (fussel178) [reply telestion-api/build.gradle.kts:14]: JUnit uses the method names for test cases by default. To make the test names more readable, we override the test name generator. (e.g. `shouldCallExtendedHandlerOnValidBuffer` maps to `should Call Extended Handler On Valid Buffer`)

**Pull request merged**